### PR TITLE
fix(docs): use correct entry for client_errors

### PR DESCRIPTION
**Pull request by kylo252**
- Point to the correct `vim.lsp.rpc.client_errors`
- ~~Add missing docs for `vim.lsp.buf_detach_client`~~

### CODE REVIEW

File: runtime/lua/vim/lsp/rpc.lua [hunk 1]
```diff
@@ -133,7 +133,8 @@ local function request_parser_loop()
   end
 end
 
-local client_errors = vim.tbl_add_reverse_lookup {
+--- Mapping of error codes used by the client
+local client_errors = {
```
Comment (kylo252) [inline runtime/lua/vim/lsp/rpc.lua:137]: I wanted to add an annotation here, but it kept it getting squashed with the next section for `format_rpc_error`.

<details>
<summary>expand diff</summary>

```diff
diff --git a/runtime/doc/lsp.txt b/runtime/doc/lsp.txt
index 1b7d727df..7ddad7ca1 100644
--- a/runtime/doc/lsp.txt
+++ b/runtime/doc/lsp.txt
@@ -1825,7 +1825,10 @@ should_log({level})                                 *vim.lsp.log.should_log()*
 Lua module: vim.lsp.rpc                                              *lsp-rpc*
 
 format_rpc_error({err})                       *vim.lsp.rpc.format_rpc_error()*
-                Constructs an error message from an LSP error object.
+                Mapping of error codes used by the client
+
+                @type table<string, number> Constructs an error message from
+                an LSP error object.
 
                 Parameters: ~
                     {err}  (table) The error object
diff --git a/runtime/lua/vim/lsp/rpc.lua b/runtime/lua/vim/lsp/rpc.lua
index 1ecac50df..f36965bce 100644
--- a/runtime/lua/vim/lsp/rpc.lua
+++ b/runtime/lua/vim/lsp/rpc.lua
@@ -134,6 +134,8 @@ local function request_parser_loop()
 end
 
 --- Mapping of error codes used by the client
+---
+---@type table<string, number>
 local client_errors = {
   INVALID_SERVER_MESSAGE       = 1;
   INVALID_SERVER_JSON          = 2;
```
</details>

File: runtime/doc/lsp.txt [hunk 2]
```diff
@@ -485,6 +485,16 @@ buf_attach_client({bufnr}, {client_id})          *vim.lsp.buf_attach_client()*
...
+buf_detach_client({bufnr}, {client_id})          *vim.lsp.buf_detach_client()*
+                Detaches client from the specified buffer. Note: While the
+                server is notified that the text document (buffer) was closed,
+                it is still able to send notifications should it ignore this
+                notification.
+
+                Parameters: ~
+                    {bufnr}      number Buffer handle, or 0 for current
+                    {client_id}  number Client id
+
```
Comment (gpanders) [inline runtime/doc/lsp.txt:497]: These doc updates were just merged earlier today: https://github.com/neovim/neovim/pull/16742

Comment (kylo252) [reply runtime/doc/lsp.txt]: Rebased!

**Comment by dundargoc** [comment]
Thank you for your contribution! We collect all typo fixes into a single pull request and merge it once it gets big enough: https://github.com/neovim/neovim/pull/16506

**Pull request closed without merging**